src/Engines/thinEngine.ts

```diff
@@ -4835,4 +4836,20 @@ export class ThinEngine {
 
         return document;
     }
+
+    /**
+     * Create a canvas
+     * @param width width
+     * @param height height
+     * @return ICanvas interface
+     */
+    public createCanvas(width: number, height: number) : ICanvas {
+        if (typeof document === "undefined") {
+            return <ICanvas>(<any>(new OffscreenCanvas(width, height)));
+        }
+        let canvas = <ICanvas>(<any>(document.createElement("canvas")));
+        canvas.width = width;
+        canvas.height = height;
+        return canvas;
+    }

```

Comment:
Do you need to remove canvasGenerator.ts?

Reply:
I did that a month ago, what I remember is cyclic dependency and I had to move it there. I don't remember the details.

Reply:
Right now there is duplicate code in canvasGenerator.ts right? So something needs to happen. :-)

Reply:
I removed canvasGenerator.ts